lib/vyasa_web/live/media_live/media_bridge.ex

```diff
@@ -69,15 +69,14 @@ defmodule VyasaWeb.MediaLive.MediaBridge do
     now = DateTime.utc_now()
     played_at = cond do
       elapsed > 0 -> # resume case
-        DateTime.add(now, -elapsed, :second)
+        DateTime.add(now, -elapsed, :millisecond)
```

Comment:
## New Bug Introduced Here

So, this is perplexing. 


Recreation Steps: 
1. use the progress bar to seek to a new time and the thing crashes 


**I'm clearly calling DateTime.add/3 here but the crash dump implies that DateTime.add/4 is being called here instead.** 
[
Based on time_unit(), `:millisecond` should be a valid option here](https://hexdocs.pm/elixir/main/System.html#t:time_unit/0)


Any idea on what this may be? Surely line 69's DateTime.utc_now() is relevant as to why it's being interpreted as DateTime.add/4 instead of DateTime.add/3



<details><summary>Here's the Crash Dump</summary>

```text

[handleEvent] seekToMs 117739.01763046545 ORIGINATOR = ProgressBar
[debug] HANDLE EVENT "seekTime" in VyasaWeb.MediaLive.MediaBridge
  Parameters: %{"originator" => "ProgressBar", "seekToMs" => 117739.01763046545}
[error] GenServer #PID<0.7885.0> terminating
** (FunctionClauseError) no function clause matching in DateTime.add/4
    (elixir 1.16.2) lib/calendar/datetime.ex:1624: DateTime.add(~U[2024-03-25 03:36:10.807997Z], -117739.01763046545, :millisecond, Calendar.UTCOnlyTimeZoneDatabase)
    (vyasa 0.1.0-alpha.1) lib/vyasa_web/live/media_live/media_bridge.ex:109: VyasaWeb.MediaLive.MediaBridge.update_playback_on_seek/2
    (vyasa 0.1.0-alpha.1) lib/vyasa_web/live/media_live/media_bridge.ex:173: VyasaWeb.MediaLive.MediaBridge.handle_seek/3
    (phoenix_live_view 0.20.1) lib/phoenix_live_view/channel.ex:487: anonymous fn/3 in Phoenix.LiveView.Channel.view_handle_event/3
    (telemetry 1.2.1) /Users/ritesh/Projects/vyasa/deps/telemetry/src/telemetry.erl:321: :telemetry.span/3
    (phoenix_live_view 0.20.1) lib/phoenix_live_view/channel.ex:246: Phoenix.LiveView.Channel.handle_info/2
    (stdlib 5.2) gen_server.erl:1095: :gen_server.try_handle_info/3
    (stdlib 5.2) gen_server.erl:1183: :gen_server.handle_msg/6
    (stdlib 5.2) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
Last message: %Phoenix.Socket.Message{topic: "lv:MediaBridge", event: "event", payload: %{"event" => "seekTime", "type" => "hook", "value" => %{"originator" => "ProgressBar", "seekToMs" => 117739.01763046545}}, ref: "15", join_ref: "7"}
State: %{socket: #Phoenix.LiveView.Socket<id: "MediaBridge", endpoint: VyasaWeb.Endpoint, view: VyasaWeb.MediaLive.MediaBridge, parent_pid: nil, root_pid: #PID<0.7885.0>, router: VyasaWeb.Router, assigns: %{session: %{"id" => "eYmEB9+/q8/Tv3km5tjEA1bo"}, __changed__: %{}, flash: %{}, locale: "en-GB", live_action: :index, tz: %{timezone: "America/Toronto", timezone_offset: -240}, should_show_vid: true, is_follow_mode: true, playback: %Vyasa.Medium.Playback{medium: nil, meta: %Vyasa.Medium.Meta{title: nil, artists: ["artistA", "artistB"], duration: 581721, file_path: "http://localhost:9000/vyasa/voices/efb211f8-9c91-453c-ab65-ead2eafc10c5.mp3?ContentType=application%2Foctet-stream&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=secrettunnel%2F20240325%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Date=20240325T033548Z&X-Amz-Expires=88888&X-Amz-SignedHeaders=host&X-Amz-Signature=4149973267895c348f17da87c4bb784139656924c4c1c0175410d6de8a36ce66"}, playing?: true, played_at: ~U[2024-03-25 03:30:28.648150Z], paused_at: nil, elapsed: 332000, current_time: 0}, video: %Vyasa.Medium.Video{__meta__: #Ecto.Schema.Metadata<:loaded, "videos">, id: "a30876ef-f031-4023-9e89-73af7c85f82e", type: "youtube", ext_uri: "AETFvQonfV8", voice_id: "efb211f8-9c91-453c-ab65-ead2eafc10c5", voice: #Ecto.Association.NotLoaded<association :voice is not loaded>, inserted_at: ~U[2024-03-05 22:22:23Z], updated_at: ~U[2024-03-05 22:22:23Z]}, voice: %Vyasa.Medium.Voice{__meta__: #Ecto.Schema.Metadata<:loaded, "voices">, id: "efb211f8-9c91-453c-ab65-ead2eafc10c5", lang: "sa", title: nil, duration: 581721, file_path: "http://localhost:9000/vyasa/voices/efb211f8-9c91-453c-ab65-ead2eafc10c5.mp3?ContentType=application%2Foctet-stream&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=secrettunnel%2F20240325%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Date=20240325T033548Z&X-Amz-Expires=88888&X-Amz-SignedHeaders=host&X-Amz-Signature=4149973267895c348f17da87c4bb784139656924c4c1c0175410d6de8a36ce66", meta: nil, video: #Ecto.Association.NotLoaded<association :video is not loaded>, events: [%Vyasa.Medium.Event{__meta__: #Ecto.Schema.Metadata<:loaded, "events">, id: "30a951b2-eeca-4e9e-a873-db83d7833144", origin: 0, duration: 2000, phase: "start", fragments: [], verse_id: nil, verse: #Ecto.Association.NotLoaded<association :verse is not loaded>, voice_id: "efb211f8-9c91-453c-ab65-ead2eafc10c5", voice: #Ecto.Association.NotLoaded<association :voice is not loaded>, source_id: "ea854925-3ddc-47a5-ab27-ab95d7b1ed24", source:
 #Ecto.Association.NotLoaded<association :source is not loaded>}, %Vyasa.Medium.Event{__meta__: #Ecto.Schema.Metadata<:loaded, "events">, id: "0a1a729a-1969-4426-b8ff-ac27353f2bd6", origin: 2000, duration: 22000, phase: nil, fragments: [], verse_id: "98591c78-d457-4312-83ca-7f0501d08b74", verse: #Ecto.Association.NotLoaded<association :verse is not loaded>, voice_id: "efb211f8-9c91-453c-ab65-ead2eafc10c5", voice: #Ecto.Association.NotLoaded<association :voice is not loaded>, source_id: "ea854925-3ddc-47a5-ab27-ab95d7b1ed24", source: #Ecto.Association.NotLoaded<association :source is not loaded>}, %Vyasa.Medium.Event{__meta__: #Ecto.Schema.Metadata<:loaded, "events">, id: "7f48dde7-bce6-4c27-9b80-c358a69e440b", origin: 24000, duration: 32000, phase: nil, fragments: [], verse_id: "99372707-ce8d-4452-85e8-55e6fd03209e", verse: #Ecto.Association.NotLoaded<association :verse is not loaded>, voice_id: "efb211f8-9c91-453c-ab65-ead2eafc10c5", voice: #Ecto.Association.NotLoaded<association :voice is not loaded>, source_id: "ea854925-3ddc-47a5-ab27-ab95d7b1ed24", source: #Ecto.Association.NotLoaded<association :source is not loaded>}, %Vyasa.Medium.Event{__meta__: #Ecto.Schema.Metadata<:loaded, "events">, id: "0fedb1cf-8945-4342-8c05-9648af1c37d9", origin: 56000, duration: 11000, phase: nil, fragments: [], verse_id: "7031f0d7-f3b5-42b0-87dd-e53958b27a3b", verse: #Ecto.Association.NotLoaded<association :verse is not loaded>, voice_id: "efb211f8-9c91-453c-ab65-ead2eafc10c5", voice: #Ecto.Association.NotLoaded<association :voice is not loaded>, source_id: "ea854925-3ddc-47a5-ab27-ab95d7b1ed24", source: #Ecto.Association.NotLoaded<association :source is not loaded>}, %Vyasa.Medium.Event{__meta__: #Ecto.Schema.Metadata<:loaded, "events">, id: "38423006-1d99-4e58-b50a-e316d9215ed6", origin: 67000, duration: 16000, phase: nil, fragments: [], verse_id: "957501db-0060-4358-8f92-2e874bfd48a8", verse: #Ecto.Association.NotLoaded<association :verse is not loaded>, voice_id: "efb211f8-9c91-453c-ab65-ead2eafc10c5", voice: #Ecto.Association.NotLoaded<association :voice is not loaded>, source_id: "ea854925-3ddc-47a5-ab27-ab95d7b1ed24", source: #Ecto.Association.NotLoaded<association :source is not loaded>}, %Vyasa.Medium.Event{__meta__: #Ecto.Schema.Metadata<:loaded, "events">, id: "24be260d-6329-4b73-8ac5-5474f7de59f7", origin: 83000, duration: 10000, phase: nil, fragments: [], verse_id: "b2f5ab9f-184d-48a7-b5b8-eccc179646b2", verse: #Ecto.Association.NotLoaded<association :verse is not loaded>, voice_id: "efb211f8-9c91-453c-ab65-ead2eafc10c5", voice: #Ecto.Association.NotLoaded<association :voice is not loaded>, source_id: "ea854925-3ddc-47a5-ab27-ab95d7b1ed24", source: #Ecto.Association.NotLoaded<association :source is not loaded>}, %Vyasa.Medium.Event{__meta__: #Ecto.Schema.Metadata<:loaded, "events">, id: "88f0aae9-f5c0-4723-bac5-6b0eaa3bc215", origin: 93000, duration: 11000, phase: nil, fragments: [], verse_id: "5bc69d4c-d11b-4f0a-ad7b-95e306cb31a0", verse: #Ecto.Association.NotLoaded<association :verse is not loaded>, voice_id: "efb211f8-9c91-453c-ab65-ead2eafc10c5", voice: #Ecto.Association.NotLoaded<association :voice is not loaded>, source_id: "ea854925-3ddc-47a5-ab27-ab95d7b1ed24", source: #Ecto.Association.NotLoaded<association :source is not loaded>}, %Vyasa.Medium.Event{__meta__: #Ecto.Schema.Metadata<:loaded, "events">, id: "066656b4-ae6b-42ee-b940-dce15cc05392", origin: 104000, duration: 11000, phase: nil, fragments: [], verse_id: "e2f9a1d6-3dc7-4dfb-bf69-0c6c2bc7585f", verse: #Ecto.Association.NotLoaded<association :verse is not loaded>, voice_id: "efb211f8-9c91-453c-ab65-ead2eafc10c5", voice: #Ecto.Association.NotLoaded<association :voice is not loaded>, source_id: "ea854925-3ddc-47a5-ab27-ab95d7b1ed24", source: #Ecto.Association.NotLoaded<association :source is not loaded>}, %Vyasa.Medium.Event{__meta__: #Ecto.Schema.Metadata<:loaded, "events">, id: "d6b57da6-5273-467a-9703-2f18d8983261", origin: 115000, duration: 15000, phase: nil, fragments: [], verse_id: "c4535596-b458-407d-b1f5-f482985a3871", verse:
 #Ecto.Association.NotLoaded<association :verse is not loaded>, voice_id: "efb211f8-9c91-453c-ab65-ead2eafc10c5", voice: #Ecto.Association.NotLoaded<association :voice is not loaded>, source_id: "ea8549 (truncated)


```


</details>

Reply:
Fixed. Bug existed because the float was too big. 

so in this calculation, use a rounded integer instead.